File: StaticDemo.java
class Demo
{
    public int i,j;         // Instance variables
    public static int k;    // Class variable

    static
    {
        System.out.println("Inside static block");
        k = 51;
    }
    public Demo()
    {
        System.out.println("Inside constructor");
        i = 11; j = 21;
    }
    public void fun()   // Instance method
    {
        System.out.println("Inside fun");
        System.out.println(i+" "+" "+j+" "+k);
    }
    public static void gun() // Class method
    {
        System.out.println("Inside gun");
        System.out.println(k);
    }
}

class StaticDemo
{
    public static void main(String A[])
    {
        System.out.println("Inside main");
        System.out.println(Demo.k);
        Demo.gun();

        Demo obj1 = new Demo();
        Demo obj2 = new Demo();
        
        System.out.println(obj1.i+" "+obj1.j);
        obj1.fun();
    }
}

File: StaticDemo.cpp
#include<iostream>
using namespace std;

class Demo
{
    public:
        int i,j,k;
        static int x;

        Demo()
        {
            i = 10; j = 20; k = 30;
        }

        void Display()
        {
            cout<<"Inside Display\n";
            cout<<i<<"\t"<<j<<"\t"<<k<<"\n";
        }

        static void Help()
        {
            cout<<"Inside Help\n";
        }
};

  
int main()
{
    Demo::Help();
    cout<<"Value of X is : "<<Demo::x<<"\n";

    Demo obj1, obj2;

    obj1.Display();
    obj2.Display();
    
    cout<<"Value of i in obj1 is : "<< obj1.i<<"\n";
    cout<<"Value of i in obj2 is : "<< obj2.i<<"\n";
    
    obj1.Help();    // Demo::Help()
    cout<<"Value of static x : "<<obj1.x<<"\n";     // Demo::x

    return 0;
}
